src/components/SampleConfigProvider.tsx

```diff
@@ -0,0 +1,167 @@
+ * string, to produce a CodeBlock with the values found in the SampleConfigContext.
+ */
+export function SampleCodeBlock(props: PropsWithChildren<{ lang: string }>) {
+    const context = useContext(SampleConfigContext);
+    const token = context.token() ? context.token() : DEFAULT_SAMPLE_TOKEN;
+    const endpoint = context.instanceKind() !== InstanceKind.SPICEDB ? AUTHZED_ENDPOINT : (context.endpoint() ?? DEFAULT_ENDPOINT);
+
+    let tenant = context.tenant() ? context.tenant() : context.defaultTenant;
+    tenant = tenant.replace('/', '');
+
+    let content = normalizeContent(props.children);
+    content = replaceAll(content, '·', '');
+
+    const template = context.buildTemplate(content);
+    const processed = template.render({ token: token, endpoint: endpoint, tenant: tenant, authzed: context.instanceKind() !== InstanceKind.SPICEDB });
+    return <CodeBlock className={`language-${props.lang}`}>{processed}</CodeBlock>
+}
+
+/**
+ * SampleConfigEditor is the editor for the values stored in the SampleConfigContext.
+ */
+export function SampleConfigEditor() {
+    const context = useContext(SampleConfigContext);
+    const instanceKind = context.instanceKind();
+    const [token, setToken] = useState(context.token() ?? '');
+    const [tenant, setTenant] = useState(context.tenant() ?? '');
+    const [endpoint, setEndpoint] = useState(context.endpoint() ?? '');
+
+    const debouncedUpdateToken = useDebouncedChecker(100, context.setToken);
+    const debouncedUpdateTenant = useDebouncedChecker(100, context.setTenant);
+    const debouncedUpdateEndpoint = useDebouncedChecker(100, context.setEndpoint);
+
+    const handleChangeToken = (e: React.ChangeEvent) => {
+        setToken(e.target.value);
+        debouncedUpdateToken.run(e.target.value);
+    };
+
+    const handleChangeTenant = (e: React.ChangeEvent) => {
+        setTenant(e.target.value);
+        debouncedUpdateTenant.run(e.target.value);
+    };
+
+    const handleChangeEndpoint = (e: React.ChangeEvent) => {
+        setEndpoint(e.target.value);
+        debouncedUpdateEndpoint.run(e.target.value);
+    };
+
+    return <div className="sample-config-editor">
+        <div className="system-options">
+            <div className="system-option">
+                <input type="radio" id="authzed" name="system" value="authzed"
+                    checked={instanceKind === InstanceKind.AUTHZED}
+                    onChange={() => context.setInstanceKind(InstanceKind.AUTHZED)} />
+                <label for="authzed">I have an <a href="https://app.authzed.com" target="_blank">Authzed permissions system</a> created</label>
+            </div>
+            <div className="system-option">
+                <input type="radio" id="spicedb" name="system" value="spicedb"
+                    checked={instanceKind === InstanceKind.SPICEDB}
+                    onChange={() => context.setInstanceKind(InstanceKind.SPICEDB)} />
```

Comment:
Need to also clear `tenant` otherwise previously set authzed prefix is displayed